commands/utils/quick-pick-azure.ts

```diff
@@ -0,0 +1,168 @@
+}
+
+export async function quickPickACRRepository(registry: Registry, prompt?: string): Promise<Repository> {
+    const placeHolder = prompt ? prompt : 'Choose registry to use';
+    const repositories: Repository[] = await acrTools.getRepositoriesByRegistry(registry);
+    const quickPickRepoList = repositories.map(repo => <IAzureQuickPickItem<Repository>>{ label: repo.name, data: repo });
+    let desiredRepo = await ext.ui.showQuickPick(quickPickRepoList, { 'canPickMany': false, 'placeHolder': placeHolder });
+    return desiredRepo.data;
+}
+
+export async function quickPickACRRegistry(canCreateNew: boolean = false, prompt?: string): Promise<Registry> {
+    const placeHolder = prompt ? prompt : 'Choose registry to use';
+    let registries = await AzureUtilityManager.getInstance().getRegistries();
+    interface RegistryWrapper {
+        isCreateNew?: boolean;
+        registry?: Registry;
+    }
+    let quickPickRegList: IAzureQuickPickItem<RegistryWrapper>[] = [];
+    if (canCreateNew) {
+        quickPickRegList.push(<IAzureQuickPickItem<RegistryWrapper>>{ label: "+ Create a new registry", data: { isCreateNew: true } });
+    }
+    quickPickRegList = quickPickRegList.concat(registries.map(reg => <IAzureQuickPickItem<RegistryWrapper>>{ label: reg.name, data: { registry: reg } }));
+    let desiredReg: IAzureQuickPickItem<RegistryWrapper> = await ext.ui.showQuickPick(quickPickRegList, {
+        'canPickMany': false,
+        'placeHolder': placeHolder
+    });
+    let registry: Registry;
+    if (canCreateNew && desiredReg.data.isCreateNew) {
+        registry = <Registry>await vscode.commands.executeCommand("vscode-docker.create-ACR-Registry");
+    } else {
+        registry = desiredReg.data.registry;
+    }
+    return registry;
+}
+
+export async function quickPickSKU(): Promise<string> {
+    const quickPickSkuList = skus.map(sk => <IAzureQuickPickItem<string>>{ label: sk, data: sk });
+    let desiredSku: IAzureQuickPickItem<string> = await ext.ui.showQuickPick(quickPickSkuList, { 'canPickMany': false, 'placeHolder': 'Choose a SKU to use' });
+    return desiredSku.data;
+}
+
+export async function quickPickSubscription(): Promise<Subscription> {
+    const subscriptions = AzureUtilityManager.getInstance().getFilteredSubscriptionList();
+    if (subscriptions.length === 0) {
+        vscode.window.showErrorMessage("You do not have any subscriptions. You can create one in your Azure portal", "Open Portal").then(val => {
+            if (val === "Open Portal") {
+                opn('https://portal.azure.com/');
+            }
+        });
+        throw new Error('User has no azure subscriptions');
+    }
+    let quickPickSubList = subscriptions.map(sub => <IAzureQuickPickItem<Subscription>>{ label: sub.displayName, data: sub });
+    let desiredSub = await ext.ui.showQuickPick(quickPickSubList, {
+        'canPickMany': false,
+        'placeHolder': 'Choose a subscription to use'
+    });
+    return desiredSub.data;
```

Comment:
If there's only a single filtered subscription, don't ask the user, just use it.